onnxruntime/core/graph/model.cc

```diff
@@ -87,17 +88,31 @@ Model::Model(const std::string& graph_name,
     opset_id_proto->set_version(domain.second);
   }
 
-  std::vector<const ONNX_NAMESPACE::FunctionProto*> model_functions;
   for (auto& func : model_local_functions) {
     auto func_ptr = model_proto_.add_functions();
     func_ptr->CopyFrom(func);
-    model_functions.emplace_back(func_ptr);
+    model_local_functions_[function_utils::GetFunctionIdentifier(func_ptr->domain(), func_ptr->name())] = func_ptr;
+  }
+
+  for (auto& func : model_proto_.functions()) {
+    auto func_schema_ptr = function_utils::CreateSchema(func.domain(),
+                                                        func.name(),
+                                                        model_local_functions_,
+                                                        *p_domain_to_version,
+                                                        *schema_registry,
+                                                        logger,
+                                                        allow_released_opsets_only);
+    auto func_template_ptr = std::make_unique<FunctionTemplate>();
+    func_template_ptr->op_schema_ = std::move(func_schema_ptr);
+    func_template_ptr->onnx_func_proto_ = &func;
+    model_local_function_templates_.push_back(std::move(func_template_ptr));
+    model_local_function_templates_maps_[function_utils::GetFunctionIdentifier(func.domain(), func.name())] = model_local_function_templates_.back().get();
```

Comment:
>model_local_function_templates_maps_ [](http://example.com/codeflow?start=4&length=36)

Since the number of entries is known, can we call reserve() on the map?